### PR TITLE
Study validation & formatting (WIP)

### PR DESCRIPTION
**Pull request by joshmoore**
This branch investigates how to validate the metadata associated with a study and additionally how to format the metadata according to different styles (primarily here for the Human Cell Atlas (HCA)).

 * [ ] Add a unit test to .travis.yml which parses a .screen file for one idr0016 study ([eg](https://github.com/IDR/idr-metadata/blob/master/idr0016-wawer-bioactivecompoundprofiling/screens/20585.screen)) The output json should be added to the repository for sharing.
 * [ ] Downloading Bio-Formats and/or OMERO (IDR-version) in .travis.yml in order to datasets which do *not* contain .screen files.

### CODE REVIEW

File: pyidr/study_parser.py [hunk 1]
```diff
@@ -308,35 +324,143 @@ def check(obj):
...
+                    {
+                        "format": "load-from-OMERO",
+                        "name": "load-from-OMERO",
+                    },
+                ]
+            }
+        print json.dumps(rv, indent=4, sort_keys=True)
+
+
+class TextPrinter(object):
```
Comment (sbesson) [inline pyidr/study_parser.py]: As mentioned separately, the object of this `TextPrinter was some initial console debugging while working on the study parser utility. Unless someone has a concrete need for it, I would not object to dropping it completely in favor of the generic JSON output.

Comment (joshmoore) [reply pyidr/study_parser.py]: Fine by me. @dominikl, @pwalczysko ?

**Comment by joshmoore** [comment]
Pushed, @sbesson

File: pyidr/study_parser.py [hunk 2]
```diff
@@ -308,35 +325,109 @@ def check(obj):
...
+            for o in objects:
+                log.info("Check annotations for %s" % o.name)
+                check(o)
+        printer.finish()
+    return p
 
 
 if __name__ == "__main__":
-    main(sys.argv[1:])
+    parser = main(sys.argv[1:])
```
Comment (sbesson) [inline pyidr/study_parser.py]: What the expected scope of returning the parser? How should it behave when multiple study files are supplied?

**Pull request merged**